packages/ra-ui-materialui/src/layout/Title.tsx

```diff
@@ -8,39 +8,41 @@ import { RaRecord, warning } from 'ra-core';
 import { PageTitleConfigurable } from './PageTitleConfigurable';
 
 export const Title = (props: TitleProps) => {
-    const { defaultTitle, title, preferenceKey, ...rest } = props;
-    const [container, setContainer] = useState(() =>
-        typeof document !== 'undefined'
-            ? document.getElementById('react-admin-title')
-            : null
-    );
+  const { defaultTitle, title, preferenceKey, nonconfigurable, ...rest } = props;
+  const [container, setContainer] = useState(() =>
+    typeof document !== 'undefined'
+      ? document.getElementById('react-admin-title')
+      : null
+  );
 
-    // on first mount, we don't have the container yet, so we wait for it
-    useEffect(() => {
-        setContainer(container => {
-            const isInTheDom =
-                typeof document !== 'undefined' &&
-                document.body.contains(container);
-            if (container && isInTheDom) return container;
-            return typeof document !== 'undefined'
-                ? document.getElementById('react-admin-title')
-                : null;
-        });
-    }, []);
+  // on first mount, we don't have the container yet, so we wait for it
+  useEffect(() => {
+    setContainer((container) => {
+      const isInTheDom =
+        typeof document !== 'undefined' && document.body.contains(container);
+      if (container && isInTheDom) return container;
+      return typeof document !== 'undefined'
+        ? document.getElementById('react-admin-title')
+        : null;
+    });
+  }, []);
 
-    if (!container) return null;
+  if (!container || nonconfigurable) return null; // Return null if nonconfigurable prop is true
```

Comment:
I don't understand why you would return `null` in this case, because then you would simply have no title at all. To me you should rather use `nonConfigurable` to choose between returning `<PageTitleConfigurable>` and `<PageTitle>`.